### PR TITLE
Convert GraphQL tests to snapshot tests

### PR DESCRIPTION
**Pull request by andrewlock**
## Summary of changes
Converts GraphQL tests to snapshots

## Reason for change

Originally started by @pierotibou in #2670 as part of an investigation, but required by #2632 to investigate differences in GraphQL v5.

## Implementation details

Switched to using Verify instead of the "manual" span expectations we were previously using. Cleaned up some now-unused files.

## Test coverage

As before.

## Other details
Closes #2670 in favour of this one

### CODE REVIEW

File: tracer/test/snapshots/GraphQL2Tests.SubmitsTraces.verified.txt [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+    Resource: GET /graphql,
+    Service: Samples.GraphQL,
+    Type: web,
+    Tags: {
+      component: aspnet_core,
+      env: integration_tests,
+      http.method: GET,
+      http.request.headers.host: localhost:00000,
+      http.status_code: 200,
+      http.url: http://localhost:00000/graphql,
```
Comment (zacharycmontoya) [inline tracer/test/snapshots/GraphQL2Tests.SubmitsTraces.verified.txt:15]: I'm surprised that the query string is not here...I can't tell if that's an issue or not

**Comment by zacharycmontoya** [comment]
CI is complaining about the overlap in snapshots:

```
System.Exception : The prefix has already been used: /project/tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/../snapshots/GraphQL4Tests.SubmitsTraces.
This is mostly caused by a conflicting combination of VerifierSettings.DerivePathInfo(), UseMethodName.UseDirectory(), UseMethodName.UseTypeName(), and UseMethodName.UseMethodName().
If that's not the case, and having multiple identical prefixes is acceptable, then call VerifierSettings.DisableRequireUniquePrefix() to disable this uniqueness validation.
```

File: tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/GraphQLTests.cs [hunk 2]
```diff
@@ -164,102 +153,68 @@ protected async Task RunSubmitsTraces(string packageVersion = "")
                     }
                 }
 
-                var spans = graphQLValidateSpans.Concat(graphQLExecuteSpans).ToList();
-                SpanTestHelpers.AssertExpectationsMet(_expectations, spans);
+                var spans = agent.WaitForSpans(expectedSpans);
```
Comment (zacharycmontoya) [inline tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/GraphQLTests.cs:156]: Maybe we can order the returned spans by StartTime to improve snapshot stability

**Comment by andrewlock** [comment]
## Benchmarks Report :snail:

Benchmarks for #2707 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d): 
* 2 benchmarks are **faster**, with geometric mean 1.227
* All benchmarks have **the same** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`WriteAndFlushEnrichedTraces`|net472|948μs|4.28μs|18.2μs|0| 0|0|3.16&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|712μs|3.47μs|13.9μs|0| 0|0|2.57&nbsp;KB|
|#2707|`WriteAndFlushEnrichedTraces`|net472|992μs|5.81μs|54.5μs|0| 0|0|3.16&nbsp;KB|
|#2707|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|746μs|4.26μs|33.2μs|0| 0|0|2.57&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AppSecBodyBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`AllCycleSimpleBody`|net472|316ns|1.21ns|4.35ns|0.066| 0|0|417&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`AllCycleSimpleBody`|netcoreapp3.1|423ns|1.27ns|4.92ns|0.00581| 0|0|416&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`AllCycleMoreComplexBody`|net472|318ns|1.35ns|5.24ns|0.0622| 0|0|393&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`AllCycleMoreComplexBody`|netcoreapp3.1|414ns|1.51ns|5.64ns|0.00541| 0|0|392&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`BodyExtractorSimpleBody`|net472|480ns|2.22ns|8.31ns|0.057| 0|0|361&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`BodyExtractorSimpleBody`|netcoreapp3.1|443ns|2.36ns|12ns|0.00387| 0|0|272&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`BodyExtractorMoreComplexBody`|net472|24.6μs|119ns|492ns|1.19| 0.0119|0|7.62&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`BodyExtractorMoreComplexBody`|netcoreapp3.1|22.2μs|129ns|1.07μs|0.094| 0|0|6.75&nbsp;KB|
|#2707|`AllCycleSimpleBody`|net472|314ns|1.26ns|5.18ns|0.066| 0|0|417&nbsp;B|
|#2707|`AllCycleSimpleBody`|netcoreapp3.1|413ns|1.6ns|6ns|0.00581| 0|0|416&nbsp;B|
|#2707|`AllCycleMoreComplexBody`|net472|315ns|1.66ns|8.64ns|0.0621| 0|0|393&nbsp;B|
|#2707|`AllCycleMoreComplexBody`|netcoreapp3.1|407ns|2.1ns|10.1ns|0.00551| 0|0|392&nbsp;B|
|#2707|`BodyExtractorSimpleBody`|net472|443ns|2.23ns|12ns|0.0569| 0|0|361&nbsp;B|
|#2707|`BodyExtractorSimpleBody`|netcoreapp3.1|438ns|2.36ns|12.9ns|0.00365| 0|0|272&nbsp;B|
|#2707|`BodyExtractorMoreComplexBody`|net472|24.7μs|136ns|829ns|1.18| 0.0121|0|7.62&nbsp;KB|
|#2707|`BodyExtractorMoreComplexBody`|netcoreapp3.1|21.9μs|126ns|979ns|0.0953| 0|0|6.75&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`SendRequest`|netcoreapp3.1|296μs|1.26μs|4.71μs|0.143| 0|0|19.74&nbsp;KB|
|#2707|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|#2707|`SendRequest`|netcoreapp3.1|299μs|1.38μs|7.7μs|0.145| 0|0|19.74&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Faster :tada: Same allocations :heavy_check_mark:



#### Faster :tada: in #2707

|  Benchmark         | base/diff | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.DbCommandBenchmark.ExecuteNonQuery&#8209;net472 | 1.203 | 2,239.68 | 1,862.18 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`ExecuteNonQuery`|net472|2.24μs|11.3ns|49.4ns|0.0938| 0|0|594&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`ExecuteNonQuery`|netcoreapp3.1|1.5μs|7.03ns|27.2ns|0.00888| 0|0|632&nbsp;B|
|#2707|`ExecuteNonQuery`|net472|1.87μs|10.5ns|69ns|0.0935| 0|0|594&nbsp;B|
|#2707|`ExecuteNonQuery`|netcoreapp3.1|1.53μs|7.62ns|36.6ns|0.00839| 0|0|632&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`CallElasticsearch`|net472|2.77μs|15.9ns|120ns|0.125| 0|0|802&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`CallElasticsearch`|netcoreapp3.1|1.81μs|9.39ns|46ns|0.0107| 0|0|792&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`CallElasticsearchAsync`|net472|2.98μs|12ns|45ns|0.149| 0.00148|0|939&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`CallElasticsearchAsync`|netcoreapp3.1|1.84μs|8.07ns|31.3ns|0.0127| 0|0|912&nbsp;B|
|#2707|`CallElasticsearch`|net472|2.67μs|14.5ns|78.3ns|0.126| 0|0|802&nbsp;B|
|#2707|`CallElasticsearch`|netcoreapp3.1|1.89μs|10.6ns|69.5ns|0.011| 0|0|792&nbsp;B|
|#2707|`CallElasticsearchAsync`|net472|3.21μs|17.2ns|92.7ns|0.148| 0.00158|0|939&nbsp;B|
|#2707|`CallElasticsearchAsync`|netcoreapp3.1|1.82μs|10.2ns|69.8ns|0.0121| 0|0|912&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Faster :tada: Same allocations :heavy_check_mark:



#### Faster :tada: in #2707

|  Benchmark         | base/diff | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.GraphQLBenchmark.ExecuteAsync&#8209;net472 | 1.251 | 3,818.05 | 3,051.55 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`ExecuteAsync`|net472|3.81μs|19.9ns|93.4ns|0.167| 0.00177|0|1.06&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`ExecuteAsync`|netcoreapp3.1|2.12μs|11.6ns|90.7ns|0.0138| 0|0|1.03&nbsp;KB|
|#2707|`ExecuteAsync`|net472|3.07μs|13.4ns|52ns|0.168| 0.0017|0|1.06&nbsp;KB|
|#2707|`ExecuteAsync`|netcoreapp3.1|2.09μs|11.1ns|58.5ns|0.0147| 0|0|1.03&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`SendAsync`|net472|7.23μs|33.9ns|140ns|0.361| 0|0|2.28&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`SendAsync`|netcoreapp3.1|4.96μs|19.8ns|74.2ns|0.0293| 0|0|2.21&nbsp;KB|
|#2707|`SendAsync`|net472|6.93μs|36.5ns|196ns|0.359| 0|0|2.28&nbsp;KB|
|#2707|`SendAsync`|netcoreapp3.1|4.96μs|16.5ns|59.4ns|0.0303| 0|0|2.21&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`EnrichedLog`|net472|3.43μs|18.7ns|106ns|0.227| 0|0|1.45&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`EnrichedLog`|netcoreapp3.1|3.22μs|18.2ns|124ns|0.0211| 0|0|1.53&nbsp;KB|
|#2707|`EnrichedLog`|net472|3.62μs|20.5ns|149ns|0.227| 0|0|1.45&nbsp;KB|
|#2707|`EnrichedLog`|netcoreapp3.1|3.28μs|17.3ns|82.8ns|0.0213| 0|0|1.53&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`EnrichedLog`|net472|290μs|1.56μs|8.54μs|0.424| 0.141|0|4.33&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`EnrichedLog`|netcoreapp3.1|232μs|1.04μs|4.04μs|0| 0|0|4.21&nbsp;KB|
|#2707|`EnrichedLog`|net472|287μs|1.45μs|6.33μs|0.423| 0.141|0|4.33&nbsp;KB|
|#2707|`EnrichedLog`|netcoreapp3.1|235μs|1.21μs|7.73μs|0| 0|0|4.21&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`EnrichedLog`|net472|7.64μs|36.1ns|140ns|0.507| 0|0|3.23&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`EnrichedLog`|netcoreapp3.1|6.92μs|37ns|248ns|0.0492| 0|0|3.6&nbsp;KB|
|#2707|`EnrichedLog`|net472|7.9μs|44.4ns|301ns|0.505| 0|0|3.23&nbsp;KB|
|#2707|`EnrichedLog`|netcoreapp3.1|6.75μs|35.5ns|170ns|0.0491| 0|0|3.6&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`SendReceive`|net472|2.39μs|13.4ns|84.6ns|0.158| 0|0|1.01&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`SendReceive`|netcoreapp3.1|2.08μs|7.63ns|33.3ns|0.0142| 0|0|1.01&nbsp;KB|
|#2707|`SendReceive`|net472|2.39μs|9.91ns|38.4ns|0.16| 0.00118|0|1.01&nbsp;KB|
|#2707|`SendReceive`|netcoreapp3.1|1.98μs|6.12ns|22.9ns|0.0141| 0|0|1.01&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`EnrichedLog`|net472|6.22μs|25.7ns|96.3ns|0.293| 0|0|1.87&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`EnrichedLog`|netcoreapp3.1|5.36μs|20.5ns|79.5ns|0.0184| 0|0|1.49&nbsp;KB|
|#2707|`EnrichedLog`|net472|6.28μs|28.1ns|116ns|0.292| 0|0|1.87&nbsp;KB|
|#2707|`EnrichedLog`|netcoreapp3.1|5.48μs|30.5ns|188ns|0.021| 0|0|1.49&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`StartFinishSpan`|net472|908ns|4.27ns|16.5ns|0.0716| 0|0|457&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`StartFinishSpan`|netcoreapp3.1|879ns|3.82ns|14.8ns|0.00615| 0|0|456&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`StartFinishScope`|net472|1.1μs|5.92ns|31.9ns|0.0846| 0|0|538&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`StartFinishScope`|netcoreapp3.1|1.06μs|5.27ns|23.6ns|0.00784| 0|0|576&nbsp;B|
|#2707|`StartFinishSpan`|net472|1.02μs|5.45ns|28.3ns|0.0715| 0|0|457&nbsp;B|
|#2707|`StartFinishSpan`|netcoreapp3.1|915ns|4.41ns|18.2ns|0.00627| 0|0|456&nbsp;B|
|#2707|`StartFinishScope`|net472|1.12μs|6.49ns|51.1ns|0.0845| 0|0|538&nbsp;B|
|#2707|`StartFinishScope`|netcoreapp3.1|1.08μs|5.85ns|34.6ns|0.00798| 0|0|576&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.TraceAnnotationsBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`RunOnMethodBegin`|net472|1.39μs|7.63ns|47.7ns|0.0838| 0|0|538&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/d7736ae9e5582ddbf4725c8b455df735c359519d)|`RunOnMethodBegin`|netcoreapp3.1|1.2μs|6.26ns|31.3ns|0.00764| 0|0|576&nbsp;B|
|#2707|`RunOnMethodBegin`|net472|1.35μs|7.58ns|59.2ns|0.0844| 0|0|538&nbsp;B|
|#2707|`RunOnMethodBegin`|netcoreapp3.1|1.19μs|6.76ns|49.2ns|0.00785| 0|0|576&nbsp;B|
</details>

File: tracer/test/snapshots/GraphQL2Tests.SubmitsTraces.verified.txt (hunk 1 above)
Comment (andrewlock) [reply tracer/test/snapshots/GraphQL2Tests.SubmitsTraces.verified.txt:15]: So... this is something I saw ages ago and then forgot - we have discrepencies over when we include the querystring and when we don't 😬 We need to look into what we should do and what we are doing in general...

File: tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/GraphQLTests.cs (hunk 2 above)
Comment (andrewlock) [reply tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/GraphQLTests.cs:156]: > Maybe we can order the returned spans by StartTime to improve snapshot stability

`VerifyHelper.VerifySpans()` does that 🙂

**Review by pierotibou — APPROVED** [review]

File: tracer/test/snapshots/GraphQL2Tests.SubmitsTraces.verified.txt (hunk 1 above)
Comment (lucaspimentel) [reply tracer/test/snapshots/GraphQL2Tests.SubmitsTraces.verified.txt:15]: Fwiw, `http.url` should _not_ include the query string by default, and users should be able to opt-in. We haven't added a setting for this in the .NET Tracer, though.

**Review by lucaspimentel — APPROVED** [review]

File: tracer/test/snapshots/GraphQL2Tests.SubmitsTraces.verified.txt (hunk 1 above)
Comment (zacharycmontoya) [reply tracer/test/snapshots/GraphQL2Tests.SubmitsTraces.verified.txt:15]: Sounds good, we can tackle this at some other time. I created a JIRA ticket to track the missing setting

**Review by zacharycmontoya — APPROVED** [review]

**Pull request merged**